sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScaleMonitor.cs

```diff
@@ -57,140 +54,7 @@ async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
-
-            return CreateTriggerMetrics(message, activeMessageCount, deadLetterCount, partitionCount, _isListeningOnDeadLetterQueue);
-        }
-
-        private async Task<ServiceBusTriggerMetrics> GetTopicMetricsAsync(ServiceBusReceivedMessage message)
-        {
-            TopicProperties topicProperties;
-            SubscriptionRuntimeProperties subscriptionProperties;
-            string topicPath, subscriptionPath;
-            long activeMessageCount = 0, deadLetterCount = 0;
-            int partitionCount = 0;
-
-            ServiceBusEntityPathHelper.ParseTopicAndSubscription(_entityPath, out topicPath, out subscriptionPath);
-
-            subscriptionProperties = await _administrationClient.Value.GetSubscriptionRuntimePropertiesAsync(topicPath, subscriptionPath).ConfigureAwait(false);
-            activeMessageCount = subscriptionProperties.ActiveMessageCount;
-            deadLetterCount = subscriptionProperties.DeadLetterMessageCount;
-
-            // If partitioning is turned on, then Service Bus automatically partitions queues into 16 partitions
-            // See more information here: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-partitioning#standard
-            topicProperties = await _administrationClient.Value.GetTopicAsync(topicPath).ConfigureAwait(false);
-            partitionCount = topicProperties.EnablePartitioning ? 16 : 0;
-
-            return CreateTriggerMetrics(message, activeMessageCount, deadLetterCount, partitionCount, _isListeningOnDeadLetterQueue);
-        }
-
-        internal static ServiceBusTriggerMetrics CreateTriggerMetrics(ServiceBusReceivedMessage message, long activeMessageCount, long deadLetterCount, int partitionCount, bool isListeningOnDeadLetterQueue)
-        {
-            long totalNewMessageCount = 0;
-            TimeSpan queueTime = TimeSpan.Zero;
-
-            if (message != null)
-            {
-                queueTime = DateTimeOffset.UtcNow.Subtract(message.EnqueuedTime);
-                totalNewMessageCount = 1; // There's at least one if message != null. Default for connection string with no manage claim
-            }
-
-            if ((!isListeningOnDeadLetterQueue && activeMessageCount > 0) || (isListeningOnDeadLetterQueue && deadLetterCount > 0))
-            {
-                totalNewMessageCount = isListeningOnDeadLetterQueue ? deadLetterCount : activeMessageCount;
-            }
-
-            return new ServiceBusTriggerMetrics
-            {
-                MessageCount = totalNewMessageCount,
-                PartitionCount = partitionCount,
-                QueueTime = queueTime
-            };
-        }
-
-        private static bool MessageIsActive(ServiceBusReceivedMessage message)
-        {
-            return message.State == ServiceBusMessageState.Active;
+            return await _serviceBusMetricsProvider.GetMetricsAsync().ConfigureAwait(false);
```

Comment:
Just curious what is purpose of ConfigureAwait(false) in this context?

Reply:
This is for performance, "Azure SDK for NET" has a rule that all awaits should be executed with `ConfigureAwait(false)` so we do not care about synchronization context (`SynchronizationContext.Current = null`)